# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/indexedDB/IDBCursor.cpp
# Commit: 9e7a90dcc429
# Full Hash: 9e7a90dcc429194fbe08e63aeed311ce100114cc
# Author: Simon Giesecke <sgiesecke@mozilla.com>
# Date: 2019-10-02 03:38:52
# Regressor Bug: 1168606
# File Overlap Count: 1
# Description:
#   Bug 1168606 - Use const where easily possible. r=ttung,asuth
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D43631
# ==============================================================================

diff -r e493a6b5f125 -r 9e7a90dcc429 dom/indexedDB/IDBCursor.cpp
--- a/dom/indexedDB/IDBCursor.cpp	Tue Oct 01 08:56:37 2019 +0000
+++ b/dom/indexedDB/IDBCursor.cpp	Tue Oct 01 09:14:04 2019 +0000
@@ -739,12 +739,9 @@
   MOZ_ASSERT(mType == Type_ObjectStore || mType == Type_Index);
   MOZ_ASSERT(!mKey.IsUnset());
 
-  IDBObjectStore* objectStore;
-  if (mType == Type_ObjectStore) {
-    objectStore = mSourceObjectStore;
-  } else {
-    objectStore = mSourceIndex->ObjectStore();
-  }
+  IDBObjectStore* const objectStore = mType == Type_ObjectStore
+                                          ? mSourceObjectStore.get()
+                                          : mSourceIndex->ObjectStore();
 
   MOZ_ASSERT(objectStore);
 